JFRWebSocket.m

```diff
@@ -717,6 +725,46 @@ -(void)dealloc
 
 /////////////////////////////////////////////////////////////////////////////
 /////////////////////////////////////////////////////////////////////////////
+@interface JFRThread ()
+
+@property(nonatomic, assign)BOOL isRunning;
+@property(nonatomic, strong)NSRunLoop *loop;
+
+@end
+/////////////////////////////////////////////////////////////////////////////
+@implementation JFRThread
+
+/////////////////////////////////////////////////////////////////////////////
++(NSRunLoop*)sharedLoop {
+    return [self shared].loop;
+}
+/////////////////////////////////////////////////////////////////////////////
++(JFRThread*)shared {
+    static JFRThread *manager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        manager = [[[self class] alloc] init];
+    });
+    return manager;
+}
+/////////////////////////////////////////////////////////////////////////////
+-(instancetype)init {
+    if(self = [super init]) {
+        self.isRunning = YES;
+        [self start];
+    }
+    return self;
+}
+/////////////////////////////////////////////////////////////////////////////
+-(void)main {
+    self.loop = [NSRunLoop currentRunLoop];
+    while (self.isRunning)
```

Comment:
Your project, your style... just want to note that industry best practices generally discourage blocks without braces in this age of defensive coding. It prevents many [little issues](http://embeddedgurus.com/barr-code/2014/03/apples-gotofail-ssl-security-bug-was-easily-preventable/). Innocuous and pedantic... until it isn't :wink: